docs/api/algorand/check-transaction.mdx

```diff
@@ -1,16 +1,60 @@
 # CheckTransaction
 
-Checks the status of a previously submitted transaction. Normally, transactions are mined within 1-5 seconds on the Algorand
-blockchain, but it can take a while longer if the blockchain is busy at the moment.
+:::warning
+This API endpoint is not yet implemented
+:::
+
+Checks the status of a previously submitted transaction. Normally, transactions are finalized within 1-5 seconds on the Algorand blockchain, but it can take a while longer if the blockchain is busy at the moment.
 
 
 ## Request Parameters
 
 |     Name    |                Type               | Description |
 | ----------- | --------------------------------- | ----------- |
-| `txId`      | `string`                          | The transaction ID received from the [SubmitTransaction](/docs/api/algorand/submit-transaction) method. |
+| `txId`      | `string`                          | A transaction ID received from the [SubmitTransaction](/docs/api/algorand/submit-transaction) method. |
 
 
 ## Responses
 
-This method is not yet implemented in the API.
+### Regular response
+
+|     Name    |   Type   | Description |
+| ----------- | -------- | ----------- | 
+| `status`    | `string` | The status of the checked transaction. Can be equal to "confirmed", "rejected", or "pending". |
```

Comment:
Please expand each status as the user's of this API may struggle to understand what it means:

+ pending - the transaction is not yet mined on the blockchain.
+ confirmed - the transaction was mined on the blockchain and can be considered final.
+ rejected - the transaction was rejected from being included to blockchain ledger. This may happen because of various racing conditions etc.

Actually, what happens if the user
a) creates a transaction via our API
b) spends funds from their wallet
c) confirms the transaction using our API

On c), will the API call fail? Or instead, the API call will go smoothly and `CheckTransaction` will give "rejected"?

Reply:
> Please expand each status as the user's of this API may struggle to understand what it means:
> 
> * pending - the transaction is not yet mined on the blockchain.
> * confirmed - the transaction was mined on the blockchain and can be considered final.
> * rejected - the transaction was rejected from being included to blockchain ledger. This may happen because of various racing conditions etc.
> 
> Actually, what happens if the user a) creates a transaction via our API b) spends funds from their wallet c) confirms the transaction using our API
> 
> On c), will the API call fail? Or instead, the API call will go smoothly and `CheckTransaction` will give "rejected"?

@ZitRos It will fail only if they spend too many funds (obviously), otherwise all good. The only problem right now is if they interact with the owner wallet which creates some time of check-time of use problems

Reply:
Thanks! But this didn't answer my question, does API call fails or API call succeeds while `CheckTransaction` fails?

Reply:
Sorry, I misunderstood. If the transaction is issued before the other spending is approved, then the SubmitTransaction endpoint succeeds, but then the TransactionStatus endpoint returns rejected. So I suppose the answer is that the API call will succeed (if you are referring to SubmitTransaction), while CheckTransaction returns rejected.

Reply:
Good to know!